helmfile.d/charts/gatekeeper/constraints/templates/no-cluster-deletion/constraint.yaml

```
@@ -0,0 +1,15 @@
+{{- if .Values.noClusterDeletion.enable -}}
+apiVersion: constraints.gatekeeper.sh/v1beta1
+kind: K8sNoClusterDeletion
+metadata:
+  name: elastisys-no-cluster-deletion
+spec:
+  enforcementAction: {{ .Values.noClusterDeletion.enforcementAction }}
+  match:
+    kinds:
+      - apiGroups: ["*"]
```

This can be limited right?
```suggestion
      - apiGroups: ["cluster.x-k8s.io"]
```

---

It looks like `openstackcluster` is part of `infrastructure.cluster.x-k8s.io`. Does that then have to be in it's own list entry or would `cluster.x-k8s.io` capture that as well?

---

I was thinking of going with what [Simon suggested](https://github.com/elastisys/compliantkubernetes-apps/pull/2405#pullrequestreview-2565043728) and rename the safeguard "prevent accidental deletion" which would allow adding more resources to it later. So I think keeping this as is would then make sense. What do you think?

---

I think it might still be worth narrowing it down otherwise it might trigger the webhook unnecessarily. Not sure if that is how the implementation actually works under the hood though.

---

I like the rewrite to make this more generic. But I still think it looks better to have several blocks of resource types with specific groups in this file (and as Simon says it might have some performance impact).